trl/trainer/grpo_trainer.py

```diff
@@ -395,14 +465,125 @@ def reward_func(completions, **kwargs):
 
     _tag_names = ["trl", "grpo"]
 
+    @staticmethod
+    def _get_nested_attr(obj, attr, default=None):
+        """Retrieve nested attribute from an object."""
+        attributes = attr.split(".")
+        for attribute in attributes:
+            try:
+                obj = getattr(obj, attribute)
+            except AttributeError:
+                return default
+        return obj
+
+    @staticmethod
+    def _get_from_processor_or_tokenizer(processor, key):
```

Comment:
Same here, I'd turn this into a function, easier if in the future we want to move it to utils and use it elsewhere